pyoxidizer.bzl

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2021 Radiotherapy AI Pty Ltd
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def make_exe():
+    dist = default_python_distribution(python_version = "3.9")
+
+    policy = dist.make_python_packaging_policy()
+    policy.resources_location = "filesystem-relative:lib"
+
+    python_config = dist.make_python_interpreter_config()
+    python_config.module_search_paths = ["$ORIGIN/lib"]
+    python_config.run_module = "pymedphys"
+
+    python_config.filesystem_importer = True
+    python_config.oxidized_importer = False
+    python_config.sys_frozen = True
+
+    exe = dist.to_python_executable(
+        name = "pymedphys",
+        packaging_policy = policy,
+        config = python_config,
+    )
+    exe.windows_runtime_dlls_mode = "always"
+    exe.windows_subsystem = "console"
+
+    exe.add_python_resources(exe.pip_install(["--use-feature", "in-tree-build", "-r", "requirements-cli.txt"]))
+    exe.add_python_resources(exe.pip_install(["--use-feature", "in-tree-build", "-r", "requirements-icom.txt"]))
+    exe.add_python_resources(exe.pip_install(["--use-feature", "in-tree-build", "-r", "requirements-tests.txt"]))
+    exe.add_python_resources(exe.pip_install(["--use-feature", "in-tree-build", "."]))
+
+    return exe
+
+def make_embedded_resources(exe):
+    return exe.to_embedded_resources()
+
+def make_install(exe):
+    # Create an object that represents our installed application file layout.
+    files = FileManifest()
+
+    # Add the generated executable to our install layout in the root directory.
+    files.add_python_resource(".", exe)
+
+    files.install("dist", replace = True)
+
+    return files
+
+def make_exe_installer(exe):
+    files = FileManifest()
+    files.add_python_resource(".", exe)
+
+    installer = WiXInstaller("pymedphys", "PyMedPhys-0.38.0-dev4-Setup.msi")
```

Comment:
Any way to avoid hardcoded value?

Reply:
I will need to think of something. This file isn't actually using Python, instead it is using [Skylark](https://docs.bazel.build/versions/main/skylark/language.html). I'll need to learn how to import files etc.

I won't fix the hard coding in this pass. But for maintainability it does need to be one of those things that is fixed. Worst case scenario... I could write it in with the `pymedphys dev propagate` call. But... more ideally... skylark could load from `pyproject.toml`... ... but I suspect "import toml" is out of the question.

Anyway... a can to be kicked down the road a bit.

There is talk of changing this config to Python proper:
https://github.com/indygreg/PyOxidizer/issues/444

Potentially if I wait long enough this hard-coding problem might be able to be solved for us by that :slightly_smiling_face:.